client/tracks/event.d.ts

```diff
@@ -6,8 +6,24 @@
  * @typedef {string} Event
  */
 export type Event =
+	// Frontend, shopper events – when recorded, these are prefixed with 'wcpay_'.
```

Comment:
This nice organisation could be broken by inattentive future engineers, e.g. adding stuff to the end or alphabetically. Would it be better to split into two objects or files?

Of course my preference would be to deprecate and remove, and use the `recordEvent` line as the source of truth – I don't think the constants are worth it. But that's a separate thing 😁

Reply:
> Would it be better to split into two objects or files?

I've mitigated this to some extent by defining separate `ShopperEvent` and `MerchantEvent` types in c8d776180a9c0f4742e73c7562e8e3c305e50c4a, and directly connecting these with the two relevant `recordEvent` functions in cbcb8a05f0263703f54511d2200ed90cf84d99af.

I think this helps to clarify that there are two distinct types of track events that each have different prefixes applied.

Note that the focus of this PR is to improve the code communicates "what is" (tracks have prefixes applied depending on whether shopper or merchant-facing) more than "what should be" (validate/enforce at the appropriate level), which I agree could be improved.